File: test/data/modifier_audit/async_functions.cs
using System;
using System.Threading.Tasks;

class Example {
    public async Task<string> FetchData(string url) {
        return await HttpClient.GetStringAsync(url);
    }

    public void SyncHelper() {
        Console.WriteLine("sync");
    }

    public static void StaticHelper() {
        Console.WriteLine("static");
    }

    private async Task PrivateAsync() {
        await Task.Delay(1000);
    }

    protected virtual void VirtualMethod() {
        Console.WriteLine("virtual");
    }
}

File: test/data/modifier_audit/async_functions.kt
import kotlinx.coroutines.*

suspend fun fetchData(url: String): String {
    return withContext(Dispatchers.IO) { "data" }
}

fun syncHelper(): Int {
    return 42
}

private suspend fun privateAsync(): Unit {
    delay(1000)
}

inline fun inlineHelper(block: () -> Unit) {
    block()
}
